microprofile/tests/junit5/src/main/java/io/helidon/microprofile/tests/junit5/HelidonJunitExtension.java

```diff
@@ -363,6 +365,17 @@ public <T> T interceptTestClassConstructor(Invocation<T> invocation,
         // we need to start container before the test class is instantiated, to honor @BeforeAll that
         // creates a custom MP config
         if (container == null) {
+
+            // at this early stage the class should be checked whether it is annotated with
+            // @TestInstance(TestInstance.Lifecycle.PER_CLASS) to start correctly the container
+            Class<?> testClass = extensionContext.getRequiredTestClass();
```

Comment:
There is a testClass field that is already populated with the correct value

Reply:
Fixed